src/atama/head/head.cpp

```diff
@@ -0,0 +1,576 @@
+              scan_tilt_angle = scan_top_limit;
+              break;
+          }
+
+          switch (scan_direction) {
+            case 0:
+              scan_pan_angle += scan_speed;
+              if (scan_pan_angle > scan_left_limit) {
+                scan_position = (scan_position + 1) % 3;
+                scan_direction = 1;
+              }
+              break;
+            case 1:
+              scan_pan_angle -= scan_speed;
+              if (scan_pan_angle < scan_right_limit) {
+                scan_position = (scan_position + 1) % 3;
+                scan_direction = 0;
+              }
+              break;
+          }
+
+          break;
+        }
+
+      case SCAN_VERTICAL:
+      case SCAN_HORIZONTAL:
+        {
+          if (init_scanning()) {
+            double value_change;
+            scan_speed = 0.8;
+
+            scan_pan_angle = get_pan_angle();
+            scan_tilt_angle = get_tilt_angle();
+
+            if (scan_mode == SCAN_VERTICAL) {
+              value_change = 15.0;
+            } else {
+              value_change = 30.0;
+            }
+
+            if (get_tilt_angle() < (scan_bottom_limit + scan_top_limit) / 2) {
+              scan_position = 0;
+              scan_tilt_angle -= value_change;
+            } else {
+              scan_position = 1;
+              scan_tilt_angle += value_change;
+            }
+          }
+
+          if (scan_mode == SCAN_VERTICAL) {
+            scan_pan_angle = (scan_left_limit + scan_right_limit) / 2;
+            switch (scan_position) {
+              case 0:
+                scan_tilt_angle += scan_speed;
+                if (scan_tilt_angle > scan_top_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+                break;
+              case 1:
+                scan_tilt_angle -= scan_speed;
+                if (scan_tilt_angle < scan_bottom_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+                break;
+            }
+          } else {
+            scan_tilt_angle = -60.0;
+            switch (scan_position) {
+              case 0:
+                scan_pan_angle += scan_speed;
+                if (scan_pan_angle > scan_left_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+
+                break;
+              case 1:
+                scan_pan_angle -= scan_speed;
+                if (scan_pan_angle < scan_right_limit) {
+                  scan_position = (scan_position + 1) % 2;
+                }
+
+                break;
+            }
+          }
+          break;
+        }
+    }
+
+    pan_angle = pan_center + keisan::clamp(scan_pan_angle, right_limit, left_limit);
+    tilt_angle = tilt_center + keisan::clamp(scan_tilt_angle, bottom_limit, top_limit);
+  }
+
+  // set calculation result
+  if (!joints.empty()) {
+    joints[0].set_position(pan_angle);
+    joints[1].set_position(tilt_angle);
+  }
```

Comment:
what if the `joints` length is less than 2?

Reply:
ok I will add the if guard there